metagraph/src/annotation/binary_matrix/multi_brwt/brwt.cpp

```diff
@@ -189,6 +193,99 @@ std::vector<BRWT::Column> BRWT::slice_rows(const std::vector<Row> &row_ids) cons
     return slice;
 }
 
+void BRWT::slice_columns(const std::vector<Column> &column_ids,
+                         const ColumnCallback &callback) const {
+    if (column_ids.empty())
+        return;
+
+    auto num_nonzero_rows = nonzero_rows_->num_set_bits();
+
+    // check if the column is empty
+    if (!num_nonzero_rows)
+        return;
+
+    // check whether it is a leaf
+    if (!child_nodes_.size()) {
+        // return the index column
+        for (size_t k = 0; k < column_ids.size(); ++k) {
+            callback(column_ids[k], std::move(*nonzero_rows_->copy()));
```

Comment:
Better call a const reference, so the column can be copied by the caller if it's needed, and otherwise, there is no overhead.
```suggestion
            callback(column_ids[k], *nonzero_rows_);
```